textattack/transformations/white_box/gradient_based_word_swap.py

```diff
@@ -0,0 +1,152 @@
+import torch
+
+from textattack.shared import utils
+from textattack.shared.validators import validate_model_gradient_word_swap_compatibility
+from textattack.transformations import Transformation
+
+class GradientBasedWordSwap(Transformation):
+    """ Uses the model's gradient to suggest replacements for a given word.
+        
+        Based off of HotFlip: White-Box Adversarial Examples for Text 
+            Classification (Ebrahimi et al., 2018).
+        
+            https://arxiv.org/pdf/1712.06751.pdf
+        
+        Arguments:
+            model (nn.Module): The model to attack. Model must have a 
+                `word_embeddings` matrix and `convert_id_to_word` function.
+            top_n (int): the number of top words to return at each index
+            replace_stopwords (bool): whether or not to replace stopwords
+    """
+    def __init__(self, model, top_n=1, replace_stopwords=False):
+        validate_model_gradient_word_swap_compatibility(model)
+        if not hasattr(model, 'word_embeddings'):
+            raise ValueError('Model needs word embedding matrix for gradient-based word swap')
+        if not hasattr(model, 'lookup_table'):
+            raise ValueError('Model needs lookup table for gradient-based word swap')
+        if not hasattr(model, 'zero_grad'):
+            raise ValueError('Model needs `zero_grad()` for gradient-based word swap')
+        if not hasattr(model.tokenizer, 'convert_id_to_word'):
+            raise ValueError('Tokenizer needs `convert_id_to_word()` for gradient-based word swap')
+        if not hasattr(model.tokenizer, 'pad_id'):
+            raise ValueError('Tokenizer needs `pad_id` for gradient-based word swap')
+        if not hasattr(model.tokenizer, 'oov_id'):
+            raise ValueError('Tokenizer needs `oov_id` for gradient-based word swap')
+        self.loss = torch.nn.CrossEntropyLoss()
+        self.model = model
+        self.pad_id = self.model.tokenizer.pad_id
+        self.oov_id = self.model.tokenizer.oov_id
+        self.top_n = top_n
```

Comment:
Need to add the line:  
self.replace_stopwords = replace_stopwords  
Or the printing of the attack throws an error for me at extra_str = extra_str.format(**self.\_\_dict\_\_):  
KeyError: 'replace_stopwords'